src/common/subprocess.ts

```diff
@@ -0,0 +1,66 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { print2 } from "../drivers/stdio";
+import { AgentArgs } from "../plugins/ssh-agent/types";
+import { spawn, SpawnOptionsWithoutStdio } from "node:child_process";
+
+/** Spawns a subprocess with given command, args, and options.
+ * May write content to its standard input.
+ * Stdout and stderr of the subprocess is printed to stderr in debug mode.
+ * The returned promise resolves with stdout or rejects with stderr of the subprocess. */
+export const asyncSpawn = async (
```

Comment:
This moved from ssh-agent.ts
The difference is now it returns stdout/stderr instead of exit code. This is helpful because when this function is called from the new gcloud path that gets the current user and the access token because:
1. we will print a nice error message if user is not logged in to gcloud (the stderr of the `gcloud ...` command)
2. we can return the access token and user as the result of awaiting this function